modules/infra/src/main/scala/scaladex/infra/ElasticsearchEngine.scala

```diff
@@ -302,9 +303,13 @@ class ElasticsearchEngine(esClient: ElasticClient, index: String)(implicit ec: E
   }
 
   private def gitHubStarScoring(query: Query): Query = {
-    val scorer = fieldFactorScore("githubInfo.stars")
-      .missing(0)
-      .modifier(FieldValueFactorFunctionModifier.LN2P)
+    val githubStarField = fieldAccess("githubInfo.stars", default = "1")
+    val scalaPercentageField = fieldAccess("githubInfo.scalaPercentage", default = "1")
+    val scorer = scriptScore(
+      Script(
+        script = s"Math.log(($githubStarField) * ($scalaPercentageField))"
```

Comment:
The argument to `Math.log` can't be 0, since the behaviour of `log(0)` is undefined/diverges toward negative infinity. As a result, the function here behaves as the log of:

  * `1`, if _both_ the star field and Scala percentage field is missing.
  * The value of the star field if the Scala percentage field is missing.
  * The value of the Scala percentage field if the star field is missing.
  * The value of the product of the star field and Scala percentage field if both are defined.

Does this sound reasonable?

Reply:
Yes it does.

However I wonder how this will behave during the transition period: the GithubUpdater will take more than 8 hours to run. During this period of time I would rather not penalize the projects that are not yet updated (Scala percentage is missing). So I suggest to use 100 as the default value of Scala percentage.

(I pushed the change)